runtime/interpreter/decode.go

```diff
@@ -973,12 +998,14 @@ func (d StorableDecoder) decodeType() (TypeValue, error) {
 	}
 
 	if err != nil {
-		return TypeValue{}, fmt.Errorf("invalid type encoding: %w", err)
+		return EmptyTypeValue, fmt.Errorf("invalid type encoding: %w", err)
 	}
 
-	return TypeValue{
-		Type: staticType,
-	}, nil
+	if staticType == nil {
+		return EmptyTypeValue, nil
+	}
```

Comment:
This condition is sort of redundant, It's same as calling the `NewUnmeteredTypeValue` constructor I suppose.

Reply:
https://github.com/onflow/cadence/pull/1512